src/main/java/com/mixpanel/android/mpmetrics/MPDbAdapter.java

```diff
@@ -102,6 +102,13 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
             db.execSQL(PEOPLE_TIME_INDEX);
         }
 
+        public boolean belowMemThreshold() {
```

Comment:
hmmm.. I don't get the point of not doing it in the MPDbAdapter. I think - 
1. mDatabaseFile is a private member of MPDatabaseHelper. so MPDatabaseHelper should manage the size of this guy.
2. to me, all the data base related stuff should be encapsulated in the MPDatabaseHelper and MPDbAdapter classes. and when we do the memory check, we don't care about the type/content/count or any specific attribute of the message, the only thing we care about is how much space we are using. so I don't think AnalyticsMessages should be involved.
3. AnalyticsMessages shouldn't know the logic of how the database is storing the messages. in the current code, all it does is insert new messages into the database by calling an API, without knowing the details of the database.